src/App.tsx

```diff
@@ -1,16 +1,54 @@
+// import { AboutMe } from './components/AboutMe/AboutMe.tsx';
+import { useState } from 'react';
+
 import { AboutMe } from './components/AboutMe/AboutMe.tsx';
 import { Footer } from './components/Footer/Footer.tsx';
 import { HeaderComponent } from './components/HeaderComponent/HeaderComponent.tsx';
+import { ProductsList } from './components/ProductsList/ProductsList.tsx';
+import { PRODUCTS_DATA } from './data/data.ts';
+import type { Product } from './interfaces/Product.ts';
 
 // import styles from './App.module.css';
 
+export type Pages = 'about' | 'products';
+
 function App() {
+    const localCart = localStorage.getItem('cart') ? JSON.parse(localStorage.getItem('cart')!) : [];
+
+    const [currentPage, setCurrentPage] = useState<Pages>('about');
+    const [cartData, setCartData] = useState<Array<Product> | []>(localCart);
+
+    function handleChangePage(page: Pages) {
+        setCurrentPage(page);
+    }
+
+    function handleAddToCart(item: Product) {
+        setCartData((previousCart) => {
+            const updatedCart = [...previousCart, item];
+            localStorage.setItem('cart', JSON.stringify(updatedCart));
+            return updatedCart;
+        });
+    }
+    let content;
+
+    switch (currentPage) {
```

Comment:
try to not create sch dynamic variables in component body
you can avoid switch case, use hash tables it is better readable use associative array instead

https://www.geeksforgeeks.org/what-are-associative-arrays-in-javascript/

example:
```ts
  switch(type) {
    case 'admin':
      return 'This User is Admin!'
      break
    case 'client':
      return 'This User is Client!'
      break
    case 'salesman':
      return 'This User is Salesman!'
      break
    default:
      return 'Ops, this guy doesn\'t have user profile'
  }
  
  // replace into
  
    const Users = {
    admin: 'This User is Admin!',
    client: 'This User is Client!',
    salesman: 'This User is Salesman!'
  }

  Users[type] || "" // now u can use this 
```
1 - less code
2 - better to read

Also u should move this logic out of react-component scope, it will be better if you will create a helper function